### DISCUSSION TITLE
Student projects

### DISCUSSION THREAD

**Discussion by jmpf** [Chaste/Chaste#248] [Announcements]
Discussion:
I talked to Proshanto this morning. (Student who came to Chaste workshop.)  He's after doing a 4th year project with Chaste *next* academic year.  I would be good to discuss things which would help the grant/Chaste and we don't mind waiting 18 months to see the fruits of.

[Slack Message](https://chasteteam.slack.com/archives/C3NPNB4F3/p1697194500525509)

**Comment by AlexFletcher** [comment]
One suggestion, which could be done in isolation from what's going on with the Chaste trunk is to implement a new cell-based population based on 'deformable polygons' - a bit like the immersed boundary method, but much simpler.

Some examples of this kind of model are:

- Tamulonis C, Postma M, Marlow HQ, Magie CR, de Jong J, Kaandorp J. 2011 A cell-based model of Nematostella vectensis gastrulation including bottle cell formation, invagination and zippering. Dev. Biol. 351, 217–228. (doi:10.1016/j.ydbio.2010.10.017)
- Perrone MC, Veldhuis JH, Brodland GW. 2016 Non-straight cell edges are important to invasion and engulfment as demonstrated by cell mechanics model. Biomech. Model. Mechanobiol. 15, 405–418. (doi:10.1007/s10237-015-0697-6)

See also Section 4 of our review article:
 - Fletcher AG, Cooper F, Baker RE. 2017 Mechanocellular models of epithelial morphogenesis. Phil. Trans. R. Soc. 372:20150519 (doi:10.1098/rstb.2015.0519)

**Comment by jmpf** [comment]
Suggestion 2 is that we could do something with solving the Eikonal equation via Fast Marching.  We could also attempt to solve the same equation with a non-linear PDE solver (though I haven't had success with that in the past).

This is all about using a non-Euclidean distance metric within a mesh to determine speed of propagation.  Think about a conductivity tensor where an electrical wave travels faster along the muscle fibre direction and less fast across sheets of fibres.  Overview paper to give ideas: https://pubmed.ncbi.nlm.nih.gov/22491074/  Sethian's Fast Marching:  https://en.wikipedia.org/wiki/Fast_marching_method

The disadvantages of this: 

- It is not directly related to this grant or to cell-based modelling
- It is something with was very fashionable in cardiac electrophysiology about 10 years ago, but may now be considered passée.

The advantages are:

- I have run similar projects as a summer coding challenge
- It can be done independently of Chaste and/or in Chaste.  It can also be parallelized in Chaste as I have parallelized an edge-based Dijkstra's algorithm.

**Comment by mirams** [comment]
I think eikonal is pretty widely used now in all the clinical patient specific modelling where they want a fast answer, certainly Gernot Plank has been using it a fair bit. https://pubmed.ncbi.nlm.nih.gov/28819329/

**Comment by jmpf** [comment]
Project on image to simulation:
_For example, would it be possible to take a super high resolution photo of somebody's arm and then convert the image into a 2D model of the skin cells in Chaste?_

This sounds similar to https://github.com/Chaste/Tiff2Chaste a project which has been worked on previously by Fergus and others.  It will form part of the work of the BBR grant so there is a good alignment between a project next academic year and the work which might need to happen.

Tasks (please expand)
- Improve cell segmentation by replacing or augmenting classic image processing (watershed) with some machine learning
- Integrate it better with the simulation pipeline
- Run some useful simulations e.g. use Chaste to predict future growth and then measure correctness
